Elcano_C2_LowLevel/Elcano_C2_LowLevel.ino

```diff
@@ -236,119 +263,156 @@ void setup()
+void loop() {
+  // Get the next loop start time. Note this (and the micros() counter) will
+  // roll over back to zero after they exceed the 32-bit size of unsigned long,
+  // which happens after about 70 minues of operation. If the new nextTime
+  // value is <= LOOP_TIME_US, we've rolled over.
+  nextTime = nextTime + LOOP_TIME_US;
+
+  // ToDo: Verify why this is being copied, e.g. is it to avoid having the
+  // values change while we're using them? Because they can change right here
+  // while we're copying them. Would it help to disable interrupts while we
+  // copy them? or does that just freeze them in some intermediate state?
+  // How about if we just make sure that anytime we exit the interrupt routine,
+  // the RC_elapsed values are in a valid state? Then we don't need to copy
+  // them. (This is all under the assumption that we're copying to avoid
+  // an inconsistent state.)
+  for (int i = 0; i < RC_NUM_SIGNALS; i++)
     local_results[i] = RC_elapsed[i];
-  //Print7(false, local_results);
-  //Serial.println("Processing RC data...");
+
   byte automate = processRC(local_results);
-  if (automate == 0x01) //remember to tell Pat to fix this, was = instead of ==
+  // @ToDo: Verify that this should be conditional. May be moot if it is
+  // replaced in the conversion to the new Elcano Serial protocol.
+  if (automate == 0x01)
   {
-    //Serial.println("Processing high-level data");
     processHighLevel(&Results);
   }
-  //    Print7( true, local_results);
-    
-  //Serial.println(nextTime); //Old sanity check line; testing new pin output sanity check.
-  //Serial.println("Clearing Results");
-//  Results.Clear();
-//  Results.kind = MSG_SENSOR;
-//  Results.angle_deg = TurnAngle_degx10() / 10;
-  //show_speed (&Results);
-
-  // Report how long the loop took.
-  
-  digitalWrite(27, HIGH);
-  if(nextTime < startTime || nextTime > startTime + LOOP_TIME_US) {
-    digitalWrite(33, HIGH);
-  }
-  //Serial.println(nextTime); //Old sanity check line; testing new pin output sanity check.
-  //Serial.println("Clearing Results");
+
+  // @ToDo: What is this doing?
   //Results.clear();
   Results.kind = MSG_SENSOR;
   Results.angle_deg = TurnAngle_degx10() / 10;
-  //show_speed (&Results);
-
-  //LogData(local_results, &Results);  // data for spreadsheet
-
-  calibrationTime_ms += LOOP_TIME_US;
-  straightTime_ms = (steer_control == STRAIGHT_TURN_OUT) ? straightTime_ms + LOOP_TIME_US : 0;
-  stoppedTime_ms = (throttle_control == MIN_ACC_OUT) ? stoppedTime_ms + LOOP_TIME_US : 0;
-  if (calibrationTime_ms > 40000 && straightTime_ms > 3000 && stoppedTime_ms > 3000)
-  {
-    //       int oldBrake = brake_control;
-    //       brake(MAX_BRAKE_OUT);  // put on brakes
-    CalibrateTurnAngle(16, 10);  // WARNING: No response to controls while calibrating
-    //       brake(oldBrake);       // restore brake state
-    calibrationTime_ms = 0;
+  // @ToDo: Is this working and should it be uncommented?
+  // show_speed (&Results);
```

Comment:
Ok, then let's leave it commented for now, and whatever's appropriate can be done with the speed-related change.

(Off topic:  Note that a previous update added code to compute speed from the motor phase feedback.  This value is currently not used.  We need to add Settings options to tell what speed sources are available.)